install/kubernetes/cilium/charts/nodeinit/templates/daemonset.yaml

```diff
@@ -24,7 +24,65 @@ spec:
           imagePullPolicy: IfNotPresent
           securityContext:
             privileged: true
+{{- if .Values.preStopRevert }}
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                  - "nsenter"
+                  - "-t"
+                  - "1"
+                  - "-m"
+                  - "--"
+                  - "/bin/sh"
+                  - "-c"
+                  - |
+                    #!/bin/bash
+
+                    set -o errexit
+                    set -o pipefail
+                    set -o nounset
+
+                    if stat /tmp/node-deinit.cilium.io > /dev/null 2>&1; then
+                      exit 0
+                    fi
+
+                    echo "Waiting on pods to stop..."
```

Comment:
Does this work as intended? The `preStop` hooks are executed inside the container before they are stopped.

Reply:
Yes, I tested it several times against GKE cluster. `preStop` is executed by issuing exec directly to the nodeinit container, we wait for other cilium pod containers to stop before  reverting configuration and we ensure that revert is only executed once via `/tmp/node-deinit.cilium.io`.

Reply:
Ah, I missde the `grep -v`, I read it as `grep` (without the `-v`)